src/Features/VisualBasic/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.vb

```diff
@@ -0,0 +1,115 @@
+﻿Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Formatting.Rules
+Imports Microsoft.CodeAnalysis.Simplification
+Imports Microsoft.CodeAnalysis.SyntaxTriviaList
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory
+
+<ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeRefactoringProviderNames.ExtractMethod), [Shared]>
+Partial Friend Class ConvertToInterpolatedStringRefactoringProvider
+    Inherits AbstractConvertToInterpolatedStringRefactoringProvider(Of InterpolatedStringExpressionSyntax, InvocationExpressionSyntax, ExpressionSyntax, ArgumentSyntax, LiteralExpressionSyntax)
+
+    Protected Overrides Function GetArguments(invocation As InvocationExpressionSyntax) As SeparatedSyntaxList(Of ArgumentSyntax) ?
+        Return invocation?.ArgumentList?.Arguments
+    End Function
+
+    Protected Overrides Function GetExpandedArguments(semanticModel As SemanticModel, arguments As SeparatedSyntaxList(Of ArgumentSyntax)) As ImmutableArray(Of ExpressionSyntax)
+        Dim builder = ImmutableArray.CreateBuilder(Of ExpressionSyntax)
+        For index = 1 To arguments.Count - 1
+            builder.Add(CastAndParenthesize(arguments(index).GetExpression, semanticModel))
+        Next
+
+        Return builder.ToImmutable()
+    End Function
+
+    Protected Overrides Function GetFirstArgument(arguments As SeparatedSyntaxList(Of ArgumentSyntax)) As LiteralExpressionSyntax
+        Return TryCast(arguments(0)?.GetExpression(), LiteralExpressionSyntax)
+    End Function
+
+    Protected Overrides Function GetInterpolatedString(text As String) As InterpolatedStringExpressionSyntax
+        Return CType(ParseExpression("$" + text), InterpolatedStringExpressionSyntax)
+    End Function
+
+    Protected Overrides Function GetText(arguments As SeparatedSyntaxList(Of ArgumentSyntax)) As String
+        Dim text = CType(arguments(0).GetExpression, LiteralExpressionSyntax).Token.ToString
+        ' We need to escape braces as this is an ambiguous case in interpolated strings that is 
+        ' not ambiguous in verbatim strings
+        Return text.Replace("'{'", "'{{'").Replace("'}'", "'}}'")
```

Comment:
In VB strings this is how you escape {

---
In reply to: [65441841](https://github.com/dotnet/roslyn/pull/11415#discussion_r65441841) [](ancestors = 65441841)